### PR TITLE
Fetch: MediaSource blob: URLs

### PR DESCRIPTION
**Pull request by annevk**
For https://github.com/whatwg/fetch/pull/708.

### CODE REVIEW

**Comment by annevk** [comment]
It seems that Safari throwing for MediaStream is arguably correct per https://w3c.github.io/mediacapture-main/ so we shouldn't add tests that encourage different behavior from that.

**Comment by annevk** [comment]
Removed that, should be all good again.

File: fetch/api/basic/mediasource.window.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+promise_test(t => {
+  return promise_rejects(t, new TypeError(), fetch(URL.createObjectURL(new MediaSource())));
```
Comment (foolip) [inline fetch/api/basic/mediasource.window.js]: Move constructor out of `promise_rejects` here too?

**Review by foolip — APPROVED** [review]
LGTM with nit

**Pull request merged**